From 22243515fdc0b61a8eca0fe52dfbcbcda759f43b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:29:33 -0400
Subject: [scsi] mpt: Use pci_stop_and_remove_bus_device_locked()

Message-id: <20140719042933.19438.1267.stgit@gir.stowe>
Patchwork-id: 86027
O-Subject: [RHEL7.1 PATCH 65/81] MPT / PCI: Use pci_stop_and_remove_bus_device_locked()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 64cdb4181d87900c6560f8c72d02c7ccf75c9070

commit 64cdb4181d87900c6560f8c72d02c7ccf75c9070
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Jan 10 15:27:56 2014 +0100

    MPT / PCI: Use pci_stop_and_remove_bus_device_locked()

    Race conditions are theoretically possible between the MPT PCI device
    removal and the generic PCI bus rescan and device removal that can be
    triggered via sysfs.

    To avoid those race conditions make the MPT PCI code use
    pci_stop_and_remove_bus_device_locked().

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/message/fusion/mptbase.c b/drivers/message/fusion/mptbase.c
index 767ff4d..570b18a 100644
--- a/drivers/message/fusion/mptbase.c
+++ b/drivers/message/fusion/mptbase.c
@@ -346,7 +346,7 @@ static int mpt_remove_dead_ioc_func(void *arg)
  if ((pdev == NULL))
   return -1;
 
- pci_stop_and_remove_bus_device(pdev);
+ pci_stop_and_remove_bus_device_locked(pdev);
  return 0;
 }
 
diff --git a/drivers/scsi/mpt2sas/mpt2sas_base.c b/drivers/scsi/mpt2sas/mpt2sas_base.c
index 3901edc..bde63f7 100644
--- a/drivers/scsi/mpt2sas/mpt2sas_base.c
+++ b/drivers/scsi/mpt2sas/mpt2sas_base.c
@@ -128,7 +128,7 @@ static int mpt2sas_remove_dead_ioc_func(void *arg)
   pdev = ioc->pdev;
   if ((pdev == NULL))
    return -1;
-  pci_stop_and_remove_bus_device(pdev);
+  pci_stop_and_remove_bus_device_locked(pdev);
   return 0;
 }
 
diff --git a/drivers/scsi/mpt3sas/mpt3sas_base.c b/drivers/scsi/mpt3sas/mpt3sas_base.c
index fa78506..0cf4f70 100644
--- a/drivers/scsi/mpt3sas/mpt3sas_base.c
+++ b/drivers/scsi/mpt3sas/mpt3sas_base.c
@@ -131,7 +131,7 @@ static int mpt3sas_remove_dead_ioc_func(void *arg)
  pdev = ioc->pdev;
  if ((pdev == NULL))
   return -1;
- pci_stop_and_remove_bus_device(pdev);
+ pci_stop_and_remove_bus_device_locked(pdev);
  return 0;
 }
 
-- 
1.7.1